libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Network/Network/Action/CSFSalesforceAction.m

```diff
@@ -280,7 +289,7 @@ - (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(N
 - (BOOL)shouldUpdateBaseUrl {
     // only set base URL to apiURL if baseURL is not already specified as absolute URL with it's own host
     // this check is necessary as there are salesforce URL that is content server based and not API based
-    return (!self.baseURL.scheme && !self.baseURL.host);
+    return (!self.baseURL.scheme && !self.baseURL.host) || (self.baseURL == self.cachedAPIURL);
```

Comment:
should we update the comments to reflect the change and why we did it the check.